gradle-failure-reports/src/test/java/com/palantir/gradle/failurereports/ThrowableFailureReporterTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.failurereports;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import com.palantir.gradle.failurereports.exceptions.ExceptionWithSuggestion;
+import com.palantir.gradle.failurereports.exceptions.MinimalException;
+import org.gradle.api.Task;
+import org.gradle.api.tasks.TaskState;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class ThrowableFailureReporterTest {
+    private static final String EXCEPTION_MESSAGE = "Exception on line 21 of script.sh";
+
+    @Mock
+    Task task;
+
+    @Mock
+    TaskState taskState;
+
+    @BeforeEach
+    public void before() {
+        when(task.getPath()).thenReturn("taskPath");
+        when(task.getState()).thenReturn(taskState);
+    }
+
+    @Test
+    public void getFailureReportSuggestion() {
```

Comment:
tests should be snake case english sentence descriptions of what they're testing, something like `exeception_with_suggestion_produces_useful_failure_report`